plugins/outputs/sematext/sematext.go

```diff
@@ -0,0 +1,294 @@
+	if len(s.Token) == 0 {
+		return fmt.Errorf("'token' is a required field for Sematext output")
+	}
+	if len(s.ReceiverURL) == 0 {
+		s.ReceiverURL = defaultSematextMetricsReceiverURL
+	}
+
+	var proxyURL *url.URL
+
+	if s.ProxyServer != "" {
+		var err error
+		proxyURL, err = url.Parse(s.ProxyServer)
+		if err != nil {
+			return fmt.Errorf("invalid url %s for the proxy server: %v", s.ProxyServer, err)
+		}
+	}
+
+	tlsConfig, err := s.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	s.senderConfig = &sender.Config{
+		ProxyURL:  proxyURL,
+		Username:  s.Username,
+		Password:  s.Password,
+		TLSConfig: tlsConfig,
+	}
+	s.sender = sender.NewSender(s.senderConfig)
+	s.metricsURL = s.ReceiverURL + "/write?db=metrics"
+
+	s.initProcessors()
+
+	s.serializer = serializer.NewMetricSerializer(s.Log)
+
+	s.Log.Infof("Sematext output started with Token=%s, ReceiverUrl=%s, ProxyServer=%s", s.Token, s.ReceiverURL,
+		s.ProxyServer)
+
+	return nil
+}
+
+// initProcessors instantiates all metric processors that will be used to prepare metrics/tags for sending to Sematext
+func (s *Sematext) initProcessors() {
+	// add more processors as they are implemented
+	s.metricProcessors = []processors.MetricProcessor{
+		processors.NewToken(s.Token),
+		processors.NewHost(s.Log),
+		processors.NewHandleCounter(),
+		processors.NewContainerTags(),
+		processors.NewMetricType(),
+	}
+	s.batchProcessors = []processors.BatchProcessor{
+		// rename processor has to run before metainfo processor to ensure metainfo processor uses final metric names
+		processors.NewRename(),
+		processors.NewHeartbeat(),
+		processors.NewMetainfo(s.Log, s.Token, s.ReceiverURL, s.senderConfig),
+	}
+}
+
+// Write sends metrics to Sematext backend and handles the response
+func (s *Sematext) Write(metrics []telegraf.Metric) error {
+	s.Log.Debugf("Sematext.Write() called with %d metrics in the slice", len(metrics))
+	processedMetrics, err := s.processMetrics(metrics)
+
+	if err != nil {
+		// error means the whole batch should be discarded without sending it. To achieve that, we have to return nil
+		s.Log.Errorf("error while preparing to send metrics to Sematext, the batch will be dropped: %v", err)
+		return nil
+	}
+
+	s.Log.Debugf("Preparing to send %d processed metrics", len(processedMetrics))
+
+	if len(processedMetrics) > 0 {
+		body := s.serializer.Write(processedMetrics)
+
+		s.Log.Debugf("Sending metrics to %s : %s", s.metricsURL, body)
+
+		res, err := s.sender.Request("POST", s.metricsURL, "text/plain; charset=utf-8", body)
+		if err != nil {
+			// error will happen in case of e.g. network connectivity issues; it is unrelated to response code and
+			// therefore it is OK to retry it
+			s.Log.Errorf("error while sending to %s : %s", s.metricsURL, err.Error())
+			return err
+		}
+		defer res.Body.Close()
+
+		s.Log.Debugf("Sending metrics to %s, response status code: %d", s.metricsURL, res.StatusCode)
+
+		return s.handleResponse(res)
+	}
+
+	return nil
+}
```

Comment:
I think you need a different approach here. Currently, you "misuse" the telegraf metrics to represent a Sematext specific structure by poking all that tags and fields. I'd rather suggest to define a Sematext specific metric type (e.g. `Datapoint` or whatever matches your terminology). Then iterate over the metrics and create and fill those `Datapoint`s using your processors. **But** those then should work on the defined Sematext structure and _not_ modify the telegraf metrics. Interpret it as a transformation to extract, collect, transform the metric values into something suited to be pushed to Sematext.